src/indexable.cr

```diff
@@ -809,6 +809,33 @@ module Indexable(T)
     index(offset) { |e| yield e } || raise Enumerable::NotFoundError.new
   end
 
+  # Returns the first element in the indexable for which the passed block
+  # is truthy, starting from the given *offset*.
+  #
+  # Accepts an optional parameter *if_none*, to set what gets returned if
+  # no element is found (defaults to `nil`).
+  #
+  # ```
+  # [1, 2, 3, 4].find { |i| i > 2 }        # => 3
+  # [1, 2, 3, 4].find(-1, 2) { |i| i < 2 } # => -1
+  # [1, 2, 3, 4].find(-1) { |i| i > 8 }    # => -1
+  # ```
+  def find(offset : Int = 0, if_none = nil, & : T ->)
+    offset += size if offset < 0
+    return nil if offset < 0
+    return (index(offset) { |i| yield i }).try { |i| unsafe_fetch(i) } || if_none
```

Comment:
Redundant `return`:

```suggestion

    (index(offset) { |i| yield i }).try { |i| unsafe_fetch(i) } || if_none
```

Reply:
good suggestion, I've updated to use this!